tasks/check_app_secrets.yml

```diff
@@ -0,0 +1,39 @@
+# Check expected app secrets exists. This is useful to avoid deploying an
+# application for which secrets have not been created first.
+
+- name: Does current app has secrets?
+  set_fact:
+    app_has_secrets: "{{
+      app |
+      json_query('services[*].templates[]') |
+      map('regex_search', '.*/secret.*\\.yml\\.j2$') |
+      select('string') | list | length > 0
+    }}"
+  tags: secret
+
+- name: "Get created application secrets for {{ app.name }}"
+  k8s_facts:
+    api_version: "v1"
+    namespace: "{{ project_name }}"
+    kind: Secret
+    label_selectors:
+      - app={{ app.name }}
+  register: existing_secrets
+  when: app_has_secrets
+  tags: secret
+
+- name: "Filter secrets matching expected checksum {{ app_vault_checksum }}"
```

Comment:
Not sure that `app_vault_checksum` is always matching the current application if `apps_filter` contains multiple apps. I'll improve this.

Reply:
Fixed.